### ISSUE TITLE
GPU memory unbalanced when training

### ISSUE THREAD

**Issue by BubbleSai** [open-mmlab/mmdetection#3147]
I have encountered the problem of gpu memory imbalance when I train images containing a large number of objects (about 50 ~ 600).  I also referred to the solution of [https://github.com/open-mmlab/mmdetection/issues/2665](url) to limit the size of box and specify the parameter gpu_assign_thr to 300, but it does not work in some cases, and it will greatly reduce the speed. Are there any other solutions, and other suggestions? Thanks!

**Comment by hellock** [comment]
You may reduce `gpu_assign_thr` and trade off between speed and memory, or use a GPU with larger memory if there are too many gts.

**Comment by BubbleSai** [comment]
> You may reduce `gpu_assign_thr` and trade off between speed and memory, or use a GPU with larger memory if there are too many gts.

Thanks for your reply. In fact, through further experiments, I found that the cause of OOM is that when the input image is too large, the number of anchor boxes (`bboxes`) increases significantly, which makes the matrix : [gt_bboxes, bboxes] in bbox_overlaps too large. Compared to put the calculations on the CPU, I tried to put all calculations in `assign` function on a separate GPU, but the speed is still unacceptable. In addition to adjusting anchor-related parameters, are there any other optimization methods? Thanks!
